### PR TITLE
[javadocs] Fix javadocs clearing all warnings

### PR DESCRIPTION
**Pull request by hazendaz**
Resolves all javadoc warnings to console that are a result of failure to comply with requested setup of same.

This does not resolve one left over item where the link is being redirected to jdk 20 but makes the output of the build finally in most reasonable state.

### CODE REVIEW

File: license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/document/Document.java [hunk 1]
```diff
@@ -49,26 +58,58 @@ public Document(File file, HeaderDefinition headerDefinition, String encoding, S
...
+   */
   public File getFile() {
     return file;
   }
 
+  /**
+   * Gets the file path.
+   *
+   * @return the file path
+   */
```
Comment (mathieucarbou) [inline license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/document/Document.java:83]: please remove all the javadoc not bringing any information.

File: license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/document/DocumentType.java [hunk 2]
```diff
@@ -28,87 +28,243 @@
...
+  /** The ruby. */
   RUBY("rb", HeaderType.SCRIPT_STYLE),
+
+  /** The scala. */
   SCALA("scala", HeaderType.SLASHSTAR_STYLE),
+
+  /** The scaml. */
   SCAML("scaml", HeaderType.HAML_STYLE),
+
+  /** The scss. */
```
Comment (mathieucarbou) [inline license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/document/DocumentType.java:190]: please don't add javadoc if you don't document.... This does not bring any information.

**Review by mathieucarbou — CHANGES_REQUESTED** [review]

**Comment by hazendaz** [comment]
converted to draft so these can be better written out to be useful.  the core reason was to remove the 100s of warnings at maven.  keeping in draft will avoid accidental merging until work can later be completed.

**Comment by mathieucarbou** [comment]
> converted to draft so these can be better written out to be useful. the core reason was to remove the 100s of warnings at maven. keeping in draft will avoid accidental merging until work can later be completed.

I understand the reason, but that's not a good fix ;-) This is a plugin so I would suggest either we remove the javadoc from everywhere except the plugin metadata (and convert existing doc to internal doc with `//`), or we correctly add the javadoc. And if some warnings require to be ignored at the javadoc plugin itself, then let's ignore them, globally or per file.

**Comment by hazendaz** [comment]
closing for now to prevent noise as I rebase branches locally on my end.  Will reopen once appropriate work done here.

**Pull request closed without merging**